src/map/zone.cpp

```diff
@@ -1197,6 +1197,315 @@ void CZone::CheckRegions(CCharEntity* PChar)
     PChar->m_InsideRegionID = RegionID;
 }
 
+std::string CZone::ZoneIDToString(uint16 zoneID)
```

Comment:
this ginormous switch shouldn't be needed when we can get zone from its ID and getName() on the zone already in other places.

Reply:
how are we requesting an npc for a zone that isn't loaded? that simply should not be happening, ever. Its a larger problem than missing an npc and we should bail out and say the zone isn't there for the message in such a case.

Reply:
I forgot but the zone is already baked into the id, we can get it from there with some bit magic

Reply:
17924154 is zone 24 and 58th entity out of its 0-4095 block

Reply:
I'm not gonna stand in the way of merge, but I really feel we should handle that situation (missing/wrong zone) differently so that naming it becomes a non-issue